internal/dataplane/parser/translate_httproute.go

```diff
@@ -279,33 +246,122 @@ func generateKongRoutesFromHTTPRouteRule(
 		// however in this case there must actually be some present hostnames
 		// configured for the HTTPRoute or else it's not valid.
 		if len(hostnames) == 0 {
-			return nil, fmt.Errorf("no match rules or hostnames specified")
+			return kongstate.Route{}, fmt.Errorf("no match rules or hostnames specified")
 		}
 
 		// otherwise apply the hostnames to the route
 		r.Hosts = append(r.Hosts, hostnames...)
+
 		// attach the plugins to be applied to the given route
-		if len(plugins) != 0 {
-			if r.Plugins == nil {
-				r.Plugins = make([]kong.Plugin, 0, len(plugins))
+		r.Plugins = append(r.Plugins, plugins...)
```

Comment:
Don't we want to preallocate (and check if we need to branch off with `len(plugin) != 0`) similarly as it's done below?

It seems that ( I just learned) `r.Plugins = append(r.Plugins, plugins...)` will assign `nil` to `r.Plugins` if `plugins` is an empty slice (or `nil`).

Reply:
it's newly created route. append will allocate memory as needed.